### DISCUSSION TITLE
Buttons

### DISCUSSION THREAD

**Discussion by safwanadnan** [devlikeapro/waha#604] [Q&A]
Are the buttons supported on all accounts now, or just WhatsApp business accounts? Also, can the buttons be used without templates? The buttons used to be available previously as well but got deprecated so is there any idea how long they will be supported this time or is it still unstable? Is there a way to receive webhooks for button responses?

**Comment by devlikepro** [comment]
> Are the buttons supported on all accounts

Tested with regular and business - works on both

> can the buttons be used without templates?

Sorry, what is it? Yes, you just send the payload as in https://waha.devlike.pro/docs/how-to/send-messages/#send-buttons

>  is there any idea how long they will be supported

:crossed_fingers: 
Consider implementing quick fallback to regular text messages anyway

> Is there a way to receive webhooks for button responses?

Yes, for `reply` it'll be in `body` (and inside `_data` the button.id as well)

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by TitoBahe** [reply]
Mine is actually, not working:

I just sent this curl, the payload is exactly the paylod from the docs but I changed the chatId to be mine personal number, and the session to be the one I am using:
```
curl -X POST http://localhost:3000/api/sendButtons \
-H "Content-Type: application/json" \
-d '{
  "chatId": "40723973035@c.us",
  "header": "How are you?",
  "body": "Tell us how are you please 🙏",
  "footer": "If you have any questions, please send it in the chat",
  "buttons": [
    {
      "type": "reply",
      "text": "I am good!"
    },
    {
      "type": "call",
      "text": "Call us",
      "phoneNumber": "+1234567890"
    },
    {
      "type": "copy",
      "text": "Copy code",
      "copyCode": "4321"
    },
    {
      "type": "url",
      "text": "How did you do that?",
      "url": "https://waha.devlike.pro"
    }
  ],
  "session": "xkCvZLdM9TSwxl5k36hU-clientId0"
}
```



[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by TitoBahe** [reply]
And I receive this from WAHA API in my server, No Erros and status code 201. But the buttons were not sent: 

And I got this in my WAHA API:
```

[14:44:10.146] DEBUG (NOWEBEngine/1): session:xkCvZLdM9TSwxl5k36hU-clientId0 - fetching sessions {"reqId":2,"session":"xkCvZLdM9TSwxl5k36hU-clientId0","jidsRequiringFetch":["40723973035@s.whatsapp.net"]}

[14:44:10.255] DEBUG (NOWEBEngine/1): session:xkCvZLdM9TSwxl5k36hU-clientId0 - adding device identity {"reqId":2,"session":"xkCvZLdM9TSwxl5k36hU-clientId0","jid":"40723973035@s.whatsapp.net"}

[14:44:10.255] DEBUG (NOWEBEngine/1): session:xkCvZLdM9TSwxl5k36hU-clientId0 - sending message to 2 devices {"reqId":2,"session":"xkCvZLdM9TSwxl5k36hU-clientId0","msgId":"3EB0F86972E6B516F2F6FBA9146CA8F94C768016"}

[14:44:10.263] INFO (1): request completed {"reqId":11,"req":{"id":11,"method":"POST","url":"/api/sendButtons","query":{},"params":{"0":"api/sendButtons"}},"res":{"statusCode":201},"responseTime":157}
```

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by TitoBahe** [reply]
And I cut out the "HeaderImage" part because it gives me this error:

`[14:41:40.003] WARN (NOWEBEngine/1): session:xkCvZLdM9TSwxl5k36hU-clientId0 - failed to obtain extra info {"reqId":2,"session":"xkCvZLdM9TSwxl5k36hU-clientId0","trace":"Error: Input file contains unsupported image format\n    at Sharp.metadata (/app/node_modules/sharp/lib/input.js:486:17)\n    at extractImageThumb (/app/node_modules/@adiwajshing/baileys/lib/Utils/messages-media.js:109:38)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async generateThumbnail (/app/node_modules/@adiwajshing/baileys/lib/Utils/messages-media.js:292:38)\n    at async /app/node_modules/@adiwajshing/baileys/lib/Utils/messages.js:131:68\n    at async Promise.all (index 1)\n    at async prepareWAMessageMedia (/app/node_modules/@adiwajshing/baileys/lib/Utils/messages.js:122:40)\n    at async WhatsappSessionNoWebPlus.uploadMedia (/app/dist/plus/engines/noweb/session.noweb.plus.js:29:34)\n    at async WhatsappSessionNoWebPlus.sendButtons (/app/dist/core/engines/noweb/session.noweb.core.js:470:29)"}`


[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
yeah, looks like buttons got fixed on WA side... working on thefix, not eta tho

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
